coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -3759,39 +3765,190 @@ def _internal_op_tensor_inplace_fill(context, node):
 
 
 @register_torch_op
-def index_put(context, node):
+def select_scatter(context, node):
     inputs = _get_inputs(context, node, expected=4)
     x = inputs[0]
+    updates = inputs[1]
+    dim = inputs[2].val
+    index = inputs[3]
+
+    # mb.torch_tensor_assign handles multi-dim slicing
+    # so we need to create slice specifications for all other dimensions
+    begin = [0] * x.rank
```

Comment:
Are making any assumption of `index.val` is not `None` here?
Wondering that will affect line 3784?

Reply:
No, `index` can be dynamic and is tested

Line 3784 `index.val not in (0, -x.rank)` works properly for `index.val is None`, since `None not in (0, -x.rank)` is True